ui/src/main/java/ucar/nc2/ui/simplegeom/SimpleGeomTable.java

```diff
@@ -0,0 +1,859 @@
+    public GeoGridBean(GridDatatype geogrid) {
+      this.geogrid = geogrid;
+      setName(geogrid.getFullName());
+      setDescription(geogrid.getDescription());
+      setUnits(geogrid.getUnitsString());
+
+      GridCoordSystem gcs = geogrid.getCoordinateSystem();
+      setCoordSystem(gcs.getName());
+
+      // collect dimensions
+      StringBuffer buff = new StringBuffer();
+      java.util.List dims = geogrid.getDimensions();
+      for (int j = 0; j < dims.size(); j++) {
+        ucar.nc2.Dimension dim = (ucar.nc2.Dimension) dims.get(j);
+        if (j > 0) buff.append(",");
+        buff.append(dim.getLength());
+      }
+      setShape(buff.toString());
+
+      x = getAxisName(gcs.getXHorizAxis());
+      y = getAxisName(gcs.getYHorizAxis());
+      z = getAxisName(gcs.getVerticalAxis());
+      t = getAxisName(gcs.getTimeAxis());
+      rt = getAxisName(gcs.getRunTimeAxis());
+      ens = getAxisName(gcs.getEnsembleAxis());
+
+      /* Dimension d= geogrid.getXDimension();
+      if (d != null) setX( d.getName());
+      d= geogrid.getYDimension();
+      if (d != null) setY( d.getName());
+      d= geogrid.getZDimension();
+      if (d != null) setZ( d.getName());
+
+      GridCoordSystem gcs = geogrid.getCoordinateSystem();
+
+      d= geogrid.getTimeDimension();
+      if (d != null)
+        setT( d.getName());
+      else {
+        CoordinateAxis taxis = gcs.getTimeAxis();
+        if (taxis != null) setT( taxis.getName());
+      }
+
+      CoordinateAxis1D axis = gcs.getEnsembleAxis();
+      if (axis != null)
+        setEns( axis.getDimension(0).getName());
+
+      axis = gcs.getRunTimeAxis();
+      if (axis != null)
+        setRt( axis.getDimension(0).getName()); */
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public void setName(String name) {
+      this.name = name;
+    }
+
+    public String getDescription() {
+      return desc;
+    }
+
+    public void setDescription(String desc) {
+      this.desc = desc;
+    }
+
+    public String getUnits() {
+      return units;
+    }
+
+    public void setUnits(String units) {
+      this.units = units;
+    }
+
+    public String getCoordSystem() {
+      return csys;
+    }
+
+    public void setCoordSystem(String csys) {
+      this.csys = csys;
+    }
+
+    public String getX() {
+      return x;
+    }
+
+    public String getY() {
+      return y;
+    }
+
+    public String getZ() {
+      return z;
+    }
+
+    public String getT() {
+      return t;
+    }
+
+    public String getEns() {
+      return ens;
+    }
+
+    public String getRt() {
+      return rt;
+    }
+
+    public String getShape() {
+      return dims;
+    }
+
+    public void setShape(String dims) {
+      this.dims = dims;
+    }
+
+    private String getAxisName(CoordinateAxis axis) {
+      if (axis != null)
+        return (axis.isCoordinateVariable()) ? axis.getShortName() : axis.getNameAndDimensions(false);
+      return "";
+    }
+  }
+

```

Comment:
Same question about this class: To make standalone, or to leave together?

Reply:
same as above.